### PR TITLE
Add "dmg" installation method for git-annex

### LINKED ISSUE #68: git-annex -m dmg  to install from a locally available .dmg file
artifacts on datalad/git-annex etc would eventually go away.  I might have a local .dmg file (e.g. from con/tinuous fetch).  Need to install it but it is not that trivial on OSX and `install_git_annex_dmg` already does all needed.  So I thought it would be nice to somehow to expose it. meanwhile will need to find my elderly bash script

### PR DESCRIPTION
**Pull request by jwodder**
Closes #68.

### CODE REVIEW

File: src/datalad_installer.py [hunk 1]
```diff
@@ -1798,6 +1798,41 @@ def assert_supported_system(self) -> None:
...
 
 
+@GitAnnexComponent.register_installer
+class DMGInstaller(Installer):
+    """ Installs a local ``*.dmg`` file """
+
+    NAME = "dmg"
+
+    OPTIONS = [
+        Option("-f", "--file", metavar="FILE", help="Path to local `*.dmg` to install"),
```
[the remark this answers is not in the document]
Comment (jwodder) [reply src/datalad_installer.py]: Renamed.

**Comment by yarikoptic** [comment]
Thank you @jwodder  !  
Could you please check why two Mac runs on appveyor don't run -- may be those types of machines are no longer there or smth like that?

**Comment by jwodder** [comment]
@yarikoptic "Cannot assign job to clouds/groups 'macos-pro'. All clouds within the group are either failing, offline, busy or reached build minutes quota."

**Comment by yarikoptic** [comment]
@mih could you check if you have this repo under you paid appveyor account?  we are trying to mimic scenarios as where datalad-installer used within datalad CI, but seems hitting the wall.  And datalad itself seems to be happy: https://ci.appveyor.com/project/mih/datalad/builds/39532629 .  So I guess we are still under a "free" account here, although it is "yours": https://ci.appveyor.com/project/mih/datalad-installer

or may be there is some other aspect?

**Comment by mih** [comment]
@yarikoptic you should have full access to this account too, and yes, the project is listed. There is no separation of free and paid within the account.

**Comment by yarikoptic** [comment]
thanks @mih !  it is odd then...

**Pull request merged**